core/src/main/java/org/geysermc/geyser/entity/type/living/ArmorStandEntity.java

```diff
@@ -293,9 +292,13 @@ public void setOffHand(ItemData offHand) {
     private void updateSecondEntityStatus(boolean sendMetadata) {
         // A secondary entity always has to have the offset applied, so it remains invisible and the nametag shows.
         if (!primaryEntity) return;
-        if (!isInvisible || isMarker) {
+        
+        if (!isInvisible) {
             // It is either impossible to show armor, or the armor stand isn't invisible. We good.
             setFlag(EntityFlag.INVISIBLE, false);
+        }
+
+        if (!isInvisible || isMarker) {
```

Comment:
Remove `isMarker` from this if statement and merge with the above if statement as we want to show armor for invisible marker armor stands.

Command to test:
`/summon armor_stand ~ ~2 ~ {CustomNameVisible:1b,NoGravity:1b,Passengers:[{id:"minecraft:axolotl",PersistenceRequired:1b,NoAI:1b}],ArmorItems:[{},{},{id:"minecraft:diamond_chestplate",Count:1b},{}],CustomName:'{"text":"Test"}',Invisible:1,Marker:1}`

Reply:
Can't put a comment on the exact line, but on line 368 change
`updateOffsetRequirement(true);` to `updateOffsetRequirement(!isMarker);`, so that the nametag for invisible markers are still correct with the above change.

Command to test:
`/summon armor_stand ~ ~2 ~ {CustomNameVisible:1b,NoGravity:1b,Passengers:[{id:"minecraft:axolotl",PersistenceRequired:1b,NoAI:1b}],CustomName:'{"text":"Test"}',Marker:1,Invisible:1}`